tests/signature/ed25519.rs

```diff
@@ -0,0 +1,72 @@
+//! Ed25519 tests
+
+use super::TestVector;
+use ring_compat::signature::{
+    ed25519::{Signature, SigningKey, VerifyKey, SIGNATURE_LENGTH},
+    Signature as _, Signer, Verifier,
+};
+
+#[test]
+fn sign_rfc8032_test_vectors() {
+    for vector in TEST_VECTORS {
+        let signing_key = SigningKey::from_seed(vector.sk).unwrap();
+        assert_eq!(signing_key.sign(vector.msg).as_ref(), vector.sig);
+    }
+}
+
+#[test]
+fn verify_rfc8032_test_vectors() {
+    for vector in TEST_VECTORS {
+        let verify_key = VerifyKey::new(vector.pk).unwrap();
+        let sig = Signature::from_bytes(vector.sig).unwrap();
+        assert!(verify_key.verify(vector.msg, &sig).is_ok());
+    }
+}
+
+#[test]
+fn rejects_tweaked_rfc8032_test_vectors() {
+    for vector in TEST_VECTORS {
+        let verify_key = VerifyKey::new(vector.pk).unwrap();
+
+        let mut tweaked_sig = [0u8; SIGNATURE_LENGTH];
+        tweaked_sig.copy_from_slice(vector.sig);
+        tweaked_sig[0] ^= 0x42;
+
+        let sig = Signature::from_bytes(&tweaked_sig[..]).unwrap();
+        assert!(verify_key.verify(vector.msg, &sig).is_err());
+    }
+}
+
+/// Ed25519 test vectors (from RFC 8032, converted to Rust bytestring literals)
+const TEST_VECTORS: &[TestVector] = &[
+    TestVector {
+        sk: b"\x9D\x61\xB1\x9D\xEF\xFD\x5A\x60\xBA\x84\x4A\xF4\x92\xEC\x2C\xC4\x44\x49\xC5\x69\x7B\x32\x69\x19\x70\x3B\xAC\x03\x1C\xAE\x7F\x60",
+        pk: b"\xD7\x5A\x98\x01\x82\xB1\x0A\xB7\xD5\x4B\xFE\xD3\xC9\x64\x07\x3A\x0E\xE1\x72\xF3\xDA\xA6\x23\x25\xAF\x02\x1A\x68\xF7\x07\x51\x1A",
```

Comment:
Use `hex-literal` to make these vectors more compact?

Reply:
Actually it's a bit tricky, since they're differently-sized...

Reply:
IIRC `&hex!("0123")[..]` should do the job.

Reply:
Yeah that should be fine. I guess on 1.47.0+ we can do `.as_ref()`?

Reply:
Probably, yes.